client/ayon_usd/standalone/usd/pinning/__pinning_file_generation_funcs.py

```diff
@@ -0,0 +1,303 @@
+import json
+import os
+import re
+from typing import Dict, List, Optional
+from pxr import UsdShade, Ar, Sdf
+from urllib.parse import urlparse
+
+
+def is_uri(path: str) -> bool:
+    parsed = urlparse(path)
+    return bool(parsed.scheme)
+
+
+def remove_root_from_dependency_info(
+    dependency_info: Dict[str, str], root_info: Dict[str, str]
+) -> Dict[str, str]:
+    """Removes the Ayon Machine Root from a given Dependecy info Dict
+
+    Args:
+        dependency_info: dict generated by `get_asset_dependencies()`
+        root_info: a flat dict containing str: root identifier {root[work]}, str: associated root path /path/to/root.
+        can be optained from Ayon server `Get Project Roots Overrides`
+
+    Returns: a dependency_info dict that holds key: Usd assetIdentifiers val: rootles paths as they would be returned
+    from Ayon server /resolve endpoint ('Resolve Uris' in the server Docs).
+    """
+
+    if not root_info or not dependency_info:
+        return {}
+
+    replacements = {
+        re.escape(path_portion): replacer
+        for replacer, path_portion in root_info.items()
+    }
+
+    pattern = re.compile("|".join(f"({pat})" for pat in replacements))
+
+    def _replace_match(match: re.Match):
+        for _, group in enumerate(match.groups(), start=1):
+            if group:
+                replacment = "{root[" + replacements[re.escape(group)] + "]}"
+                return replacment
+        return match.group(0)
+
+    root_less_dict = {}
+
+    for key, path in dependency_info.items():
+        new_path = pattern.sub(_replace_match, path)
+        root_less_dict[key] = new_path
+
+    return root_less_dict
+
+
+def _get_prim_prop_data(prim: Sdf.PrimSpec, layer: Sdf.Layer) -> List[str]:
+    """get all Sdf.AttributeSpec data including time sample data fomr a given Sdf.PrimSpec
```

Comment:
```suggestion
    """get all Sdf.AttributeSpec data including time sample data from a given Sdf.PrimSpec
```

Also, technically it doesn't return all AttributeSpec data, but only for Asset type Attribute Specs.

```suggestion
    """Return all Sdf.AttributeSpec values including time sample data from a Sdf.PrimSpec
    for all the Asset type Attribute Specs.
```